.github/workflows/valgrind-tests.yml

```diff
@@ -0,0 +1,29 @@
+on:
+  workflow_call:
+    inputs:
+      portion:
+        required: true
```

Comment:
I would drop the required keyword and instead add a default and a description, like so:
```
description: "Select portion of tests to run under Valgrind (default runs all):"
default: ""
```

Reply:
(assuming description and default work for workflow_call)

Reply:
We could instead have the Valgrind workflow be triggered by the successful run of the `ci.yml` workflow. It would still allow splitting the portions, but also skip Valgrind entirely if other tests fail.

Reply:
okay, sounds good to me

Reply:
> We could instead have the Valgrind workflow be triggered by the successful run of the ci.yml workflow. It would still allow splitting the portions, but also skip Valgrind entirely if other tests fail.

Let me do that in a separate PR. I first need to read through GH actions doc to learn how to do it.